reflect/reflect.go

```diff
@@ -75,36 +75,64 @@ func FieldsAndValues(val interface{}) ([]string, []interface{}, bool) {
 	return fields, values, true
 }
 
-var structMapMutex sync.RWMutex
-var structMap = make(map[r.Type]*structInfo)
+type typeMap struct {
+	m sync.RWMutex
 
-type fieldInfo struct {
-	Key string
-	Num int
+	types map[r.Type]*StructInfo
 }
 
-type structInfo struct {
+// Get returns information of a mapped struct if it has been processed by gocqltable.
+func (t *typeMap) Get(i interface{}) (*StructInfo, bool) {
+	var rt r.Type
+	switch v := i.(type) {
+	case r.Type:
+		rt = v
+	default:
+		rt = r.Indirect(r.ValueOf(i)).Type()
+	}
+	t.m.Lock()
+	s, found := t.types[rt]
+	t.m.Unlock()
+	return s, found
+}
+
+func (t *typeMap) set(rt r.Type, v *StructInfo) {
```

Comment:
The setter is only used internally so shouldn't be exposed to the outside world.